BusTracker/Networking/APIResponceParser.swift

```
@@ -9,53 +9,108 @@
-                    print(responce["error"] ?? "Smth wrong with passing an API key")
-                case "Bledna metoda":
-                    print("Wrong type requested. Must be in (1, 2)")
-                case "Błędna metoda lub parametry wywołania":
-                    print("No parameter: resource_id")
-                default:
-                    print("What's going on???")
-                }
+                
+                defineAPIError(result: message, error: responce["error"] as? String)
+                
             } else {
                 
                 let vehiclePositions = responce["result"] as? [Any] ?? []
                 
-                for vehicle in vehiclePositions {
-                    let vehicleData = vehicle as! [String : Any]
-                    
-                    let positionUpdateDatetime = dateFormatter.date(from: vehicleData["Time"] as! String)!
-                    
-                    //Create a TransportPosition object
-                    let parcedVehicle = TransportPosition(line: vehicleData["Lines"] as! String,
-                                                          latitude: vehicleData["Lat"] as! Double,
-                                                          longitude: vehicleData["Lon"] as! Double,
-                                                          brigade: vehicleData["Brigade"] as! String,
-                                                          updateTime: positionUpdateDatetime,
-                                                          vehicleType: vehicleType)
-                    
-                    parcedVehiclePositions.append(parcedVehicle)
-                }
+                parcedVehiclePositions = positionDataToArray(vehicleData: vehiclePositions, vehicleType: vehicleType)
             }
         }
         
+        //print(parcedVehiclePositions)
         return parcedVehiclePositions
     }
+    
+    func defineAPIError (result : String, error : String?) {
```

I guess this method should return error instead of printing it, so that error can be forwarded up the call hierarchy in `parcePositions` method. Or haven't I understood something here?

Also I think that naming can be improved, currently it is a bit unclear what this method does. I would rather go for
```swift
func createAPIError(from result: String, error String)
```
 or something like this.